File: 75-sort-colors/sort-colors.ts
/**
 Do not return anything, modify nums in-place instead.
 */
// function sortColors(nums: number[]): void {

function sortColors(unsortedArray:number[]) {
    let sortedArray = [];
    let current;
    for (let i = 0; i < unsortedArray.length; i++) {
        current = unsortedArray[i]
        for (let j = i + 1; j < unsortedArray.length; j++) {

            if (unsortedArray[j] < current) {
                current = unsortedArray[j];
                unsortedArray[j] = unsortedArray[i];
                unsortedArray[i] = current;
            }

        }
        sortedArray.push(current);

    }

    return sortedArray
}

// };

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        lf,rt=0,len(nums)-1
        i=0
        def swap(i,j):
            temp=nums[i]
            nums[i]=nums[j]
            nums[j]=temp
        
        while i <= rt:
            if nums[i]==0:
                swap(i,lf)
                lf+=1
            elif nums[i] == 2:
                swap(i,rt)
                rt-=1
                i-=1
            i+=1
        return nums


        